tests/end2end/data/fileselect/multiple_fileselect.html

```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <title>Testing fileselection</title>
+        <script type="text/javascript">
+            function onLoad() {
+                const input = document.querySelector('#file_upload');
+                input.addEventListener('input', function (e) {
+                    for (i = 0; i < input.files.length; i++) {
```

Comment:
Perhaps you could use something like

```json
console.log(`Files: ${JSON.stringify(input.files)}`)
```

here, so that you get a single list of all files? That'd then also simplify the Python side of the test a bit, and make it more robust at the same time, as you can make sure there's nothing unexpected in there.

It'd likely also allow fully sharing the code between the two variants, see my comment below.

Reply:
Sounds good. The only other difference is if the input button should have the `multiple` boolean flag or not. But maybe there can be two buttons with different ids, one for single and one for multiple but they both trigger the same JS function.

Reply:
Yep, that should work!

Reply:
I've updated this. However seems `input.files` is a more complex structure, a dictionary of objects? I haven't done much JS before but in python I would do something like `[file.name for file in input.files.values]`. What's a good way to do something similar in JS? :)